service/controller/controllercontext/spec.go

```diff
@@ -0,0 +1,24 @@
+package controllercontext
+
+import (
+	releasev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/release/v1alpha1"
+)
+
+type ContextSpec struct {
```

Comment:
I decided to add a `Spec` field to the `controllercontext` in which we can pass around the desired state of a resource. In this case, it contains the release (based on CR label) and ignition hash which should be reconciled to become the actual state.